cmake/modules/Libdispatch.cmake

```diff
@@ -0,0 +1,149 @@
+                          <INSTALL_DIR>/${LIBDISPATCH_RUNTIME_DIR}/${SWIFT_SDK_${sdk}_SHARED_LIBRARY_PREFIX}BlocksRuntime${SWIFT_SDK_${sdk}_SHARED_LIBRARY_SUFFIX}
+                          <INSTALL_DIR>/lib/${SWIFT_SDK_${sdk}_SHARED_LIBRARY_PREFIX}BlocksRuntime${SWIFT_SDK_${sdk}_SHARED_LIBRARY_SUFFIX}
+                        BUILD_ALWAYS
+                          1)
+
+    ExternalProject_Get_Property("${LIBDISPATCH_VARIANT_NAME}" install_dir)
+
+    # CMake does not like the addition of INTERFACE_INCLUDE_DIRECTORIES without
+    # the directory existing.  Just create the location which will be populated
+    # during the installation.
+    file(MAKE_DIRECTORY ${install_dir}/include)
+
+    set(DISPATCH_VARIANT_NAME "dispatch-${SWIFT_SDK_${sdk}_LIB_SUBDIR}-${arch}")
+    add_library("${DISPATCH_VARIANT_NAME}" SHARED IMPORTED GLOBAL)
+    set_target_properties("${DISPATCH_VARIANT_NAME}"
+                          PROPERTIES
+                            IMPORTED_LOCATION
+                              ${install_dir}/${LIBDISPATCH_RUNTIME_DIR}/${SWIFT_SDK_${sdk}_SHARED_LIBRARY_PREFIX}dispatch${SWIFT_SDK_${sdk}_SHARED_LIBRARY_SUFFIX}
+                            IMPORTED_IMPLIB
+                              ${install_dir}/lib/${SWIFT_SDK_${sdk}_SHARED_LIBRARY_PREFIX}dispatch${SWIFT_SDK_${sdk}_SHARED_LIBRARY_SUFFIX}
+                            INTERFACE_INCLUDE_DIRECTORIES
+                              ${install_dir}/include
+                            IMPORTED_NO_SONAME
+                              1)
+
+    set(BLOCKS_RUNTIME_VARIANT_NAME "BlocksRuntime-${SWIFT_SDK_${sdk}_LIB_SUBDIR}-${arch}")
+    add_library("${BLOCKS_RUNTIME_VARIANT_NAME}" SHARED IMPORTED GLOBAL)
+    set_target_properties("${BLOCKS_RUNTIME_VARIANT_NAME}"
+                          PROPERTIES
+                            IMPORTED_LOCATION
+                              ${install_dir}/${LIBDISPATCH_RUNTIME_DIR}/${SWIFT_SDK_${sdk}_SHARED_LIBRARY_PREFIX}BlocksRuntime${SWIFT_SDK_${sdk}_SHARED_LIBRARY_SUFFIX}
+                            IMPORTED_IMPLIB
+                              ${install_dir}/lib/${SWIFT_SDK_${sdk}_SHARED_LIBRARY_PREFIX}BlocksRuntime${SWIFT_SDK_${sdk}_SHARED_LIBRARY_SUFFIX}
+                            INTERFACE_INCLUDE_DIRECTORIES
+                              ${SWIFT_PATH_TO_LIBDISPATCH_SOURCE}/src/BlocksRuntime
+                            IMPORTED_NO_SONAME
+                              1)
+
+    add_dependencies("${DISPATCH_VARIANT_NAME}" "${LIBDISPATCH_VARIANT_NAME}-install")
+    add_dependencies("${BLOCKS_RUNTIME_VARIANT_NAME}" "${LIBDISPATCH_VARIANT_NAME}-install")
+
+    if("${sdk}" STREQUAL "${SWIFT_HOST_VARIANT_SDK}")
+      if("${arch}" STREQUAL "${SWIFT_HOST_VARIANT_ARCH}")
+        add_library(dispatch ALIAS "${DISPATCH_VARIANT_NAME}")
+        add_library(BlocksRuntime ALIAS "${BLOCKS_RUNTIME_VARIANT_NAME}")
+      endif()
+    endif()
+  endforeach()
+endforeach()
+
+if(DISPATCH_SDKS)
```

Comment:
Hmm, I suspect that this is no longer needed; however, I don't think that it matters too much, since you are adding paths to the _source_ directory, not the build directory.